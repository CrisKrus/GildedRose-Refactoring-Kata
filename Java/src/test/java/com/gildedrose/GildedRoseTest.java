package com.gildedrose;

import org.approvaltests.Approvals;
import org.junit.jupiter.api.Test;

import java.util.Arrays;

import static org.junit.jupiter.api.Assertions.assertEquals;

class GildedRoseTest {

    @Test
    void foo() {
        Item[] items = new Item[] {
            new Item("foo", 0, 0),
            new Item("foo", 5, 0),
            new Item("foo", 10, 0),
            new Item("foo", 0, 5),
            new Item("foo", 5, 5),
            new Item("foo", 10, 5),
        };
        GildedRose app = new GildedRose(items);
        app.updateQuality();

        Approvals.verifyAll("foo", Arrays.asList(items));
    }

    @Test
    void AgedBrie() {
        Item[] items = new Item[] {
            new Item("Aged Brie", 0, 0),
            new Item("Aged Brie", 5, 0),
            new Item("Aged Brie", 10, 0),
            new Item("Aged Brie", 0, 5),
            new Item("Aged Brie", 5, 5),
            new Item("Aged Brie", 10, 5),
        };
        GildedRose app = new GildedRose(items);
        app.updateQuality();

        Approvals.verifyAll("foo", Arrays.asList(items));
    }

    @Test
    void SulfurasHadnOfRagnaros() {
        Item[] items = new Item[] {
            new Item("Sulfuras, Hand of Ragnaros", 0, 0),
            new Item("Sulfuras, Hand of Ragnaros", 5, 0),
            new Item("Sulfuras, Hand of Ragnaros", 10, 0),
            new Item("Sulfuras, Hand of Ragnaros", 0, 5),
            new Item("Sulfuras, Hand of Ragnaros", 5, 5),
            new Item("Sulfuras, Hand of Ragnaros", 10, 5),
        };
        GildedRose app = new GildedRose(items);
        app.updateQuality();

        Approvals.verifyAll("foo", Arrays.asList(items));
    }

    @Test
    void BackstagePassesToATafkal80etcConcert() {
        Item[] items = new Item[] {
            new Item("Backstage passes to a TAFKAL80ETC concert", 0, 0),
            new Item("Backstage passes to a TAFKAL80ETC concert", 5, 0),
            new Item("Backstage passes to a TAFKAL80ETC concert", 10, 0),
            new Item("Backstage passes to a TAFKAL80ETC concert", 0, 5),
            new Item("Backstage passes to a TAFKAL80ETC concert", 5, 5),
            new Item("Backstage passes to a TAFKAL80ETC concert", 10, 5),
        };
        GildedRose app = new GildedRose(items);
        app.updateQuality();

        Approvals.verifyAll("foo", Arrays.asList(items));
    }

}
