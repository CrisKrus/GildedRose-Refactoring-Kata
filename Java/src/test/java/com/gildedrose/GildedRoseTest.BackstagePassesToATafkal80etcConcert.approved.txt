foo[0] = Backstage passes to a TAFKAL80ETC concert, -1, 0
foo[1] = Backstage passes to a TAFKAL80ETC concert, 4, 3
foo[2] = Backstage passes to a TAFKAL80ETC concert, 9, 2
foo[3] = Backstage passes to a TAFKAL80ETC concert, -1, 0
foo[4] = Backstage passes to a TAFKAL80ETC concert, 4, 8
foo[5] = Backstage passes to a TAFKAL80ETC concert, 9, 7
